### PR TITLE
Update arrow-flight subcrates (#3044)

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Part of #3044

# Rationale for this change
 

Updates arrow-flight to depend only on sub crates

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by tustvold** [comment]
Marking as draft pending #3053

File: .github/workflows/arrow_flight.yml [hunk 1]
```diff
@@ -36,8 +35,6 @@ on:
       - arrow-select/**
       - arrow-flight/**
       - arrow-ipc/**
-      - arrow-csv/**
-      - arrow-json/**
```
Comment (viirya) [inline .github/workflows/arrow_flight.yml:40]: Added by mistake?

File: arrow-flight/src/sql/mod.rs [hunk 2]
```diff
@@ -122,18 +122,18 @@ pub trait ProstAnyExt {
     ///
     /// * `Ok(None)` when message type mismatch
     /// * `Err` when parse failed
-    fn unpack<M: ProstMessageExt>(&self) -> ArrowResult<Option<M>>;
+    fn unpack<M: ProstMessageExt>(&self) -> Result<Option<M>, ArrowError>;
```
Comment (viirya) [inline arrow-flight/src/sql/mod.rs:125]: I saw you defined `ArrowResult` at `arrow-flight/src/lib.rs`, aren't we going to use it across arrow-flight crate?

**Review by viirya — APPROVED** [review]

File: .github/workflows/arrow_flight.yml (hunk 1 above)
Comment (tustvold) [reply .github/workflows/arrow_flight.yml:40]: At the time arrow-flight had a transitive dependency on them via arrow

File: arrow-flight/src/sql/mod.rs (hunk 2 above)
Comment (tustvold) [reply arrow-flight/src/sql/mod.rs:125]: In general I've been trying to avoid aliasing Result, it was necessary in lib.rs due to name collisions

**Pull request merged**